lib/legitbot/bing.rb

```diff
@@ -1,12 +1,10 @@
-module Legitbot
+# frozen_string_literal: true
+
+module Legitbot # :nodoc:
   # https://blogs.bing.com/webmaster/2012/08/31/how-to-verify-that-bingbot-is-bingbot/
   class Bing < BotMatch
-    ValidDomains = ["search.msn.com."]
-
-    def valid?
-      subdomain_of?(*Bing::ValidDomains) && reverse_resolves?
-    end
+    domains 'search.msn.com.'
```

Comment:
Dropping the reverse lookup seems prudent, given the PTRs records that don't actually resolve.

Reply:
`resolve` in `domains` is by default set to `true`. But `domains` validator is supposed to work around the issue you had. And the tests for this logic is present.

Reply:
Indeed, I re-did my test and it's returning correct result and not hanging for a few seconds while trying to resolve the non-existent PTR record.